include/indicators/details/stream_helper.hpp

```diff
@@ -65,5 +69,69 @@ std::ostream &print_duration(std::ostream &os, std::chrono::nanoseconds ns) {
     return os;
 };
 
+class BlockProgressScaleWriter
+{
+public:
+    BlockProgressScaleWriter(std::ostream& os, size_t bar_width)
+        : os(os)
+        , bar_width(bar_width)
+    {}
+
+    std::ostream &write(float progress) {
+        std::string fill_text{"█"};
+        std::vector<std::string> lead_characters{" ", "▏", "▎", "▍", "▌", "▋", "▊", "▉"};
+        auto value = std::min(1.0f, std::max(0.0f, progress / 100.0f));
+        auto whole_width = std::floor(value * bar_width);
+        auto remainder_width = fmod((value * bar_width), 1.0f);
+        auto part_width = std::floor(remainder_width * lead_characters.size());
+        std::string lead_text = lead_characters[size_t(part_width)];
+        if ((bar_width - whole_width - 1) < 0)
+        lead_text = "";
+        for (size_t i = 0; i < whole_width; ++i)
+        os << fill_text;
+        os << lead_text;
+        for (size_t i = 0; i < (bar_width - whole_width - 1); ++i)
+        os << " ";
+    }
```

Comment:
Return `os`

Reply:
@p-ranav I'm so sorry for my dummy mistakes. Fixed.